src/plugin/helpers/decorations.ts

```diff
@@ -0,0 +1,84 @@
+import type { Node } from "prosemirror-model";
+import { Mapping, StepMap } from "prosemirror-transform";
+import type { DecorationGroup, DecorationSource } from "prosemirror-view";
+import { DecorationSet } from "prosemirror-view";
+
+export const getMappedDecorationsFromSet = (
+  // A decoration set received, positioned relative to the outer editor. It may contain irrelevant decorations.
+  decorationSet: DecorationSet,
+  // The offset of the field from its containing element.
```

Comment:
I'm confused about this, it is the offset of the field from the entire doc, or just from the containing element?

Reply:
The comment is correct. The decorations are already offset from the beginning of the element.